ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -432,17 +434,24 @@ private void expandMapType(String jsonFieldName, Object val, Field field,
     Optional<Object> value = Optional.ofNullable(val);
     value.filter(Map.class::isInstance).map(Map.class::cast).ifPresent(m -> {
       Map<String, Object> map = m;
-      Field valueField = field.getSubFields().get(1);
       Map<String, Object> props = additionalProperties == null ? null : new HashMap<>();
+      Optional<Field> valueFieldOption;
+      if (field.getSubFields().size() == 2) {
+        valueFieldOption = Optional.of(field.getSubFields().get(1));
+      } else {
+        valueFieldOption = Optional.empty();
+        props.putAll(map);
```

Comment:
pretty sure this needs a null check:
```suggestion
        if (props != null) {
          props.putAll(map);
        }
```